client/src/pages/LogIn_SignUp.js

```diff
@@ -0,0 +1,208 @@
+import {
+  TextField,
+  Button,
+  Typography,
+  Link,
+  Paper,
+  Grid,
+  Box,
+  makeStyles,
+} from '@material-ui/core';
+import React, { useEffect, useState, useCallback } from 'react';
+import { useLocation } from 'react-router-dom';
+
+const useStyles = makeStyles({
+  textField: {
+    width: '50%',
+  },
+  bold: {
+    color: 'black',
+  },
+  seperator: {
+    width: '10%',
+    border: '1px solid #00e676',
+  },
+  formPadding: {
+    padding: '20px',
+  },
+});
+
+const LogIn_SignUp = () => {
+  //state declaration
+  const [signInEmail, setEmail] = useState('');
+  const [signInPassword, setPassword] = useState('');
+  const [registerName, setName] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [lengthValidation, setLengthValidation] = useState(true);
+  const [confirmValidation, setConfirmValidation] = useState(true);
+
+  const classes = useStyles();
+
+  //Determines which form elements to render based on the path
+  const signInForm = useLocation().pathname === '/login' ? true : false;
+
+  //change this to change the min length requirement
+  const minLength = 6;
+
+  //front-end validation for registering users
+  const handleValidation = useCallback(() => {
+    if (signInPassword !== '' && signInPassword.length < minLength) {
+      setLengthValidation(false);
+    } else {
+      setLengthValidation(true);
+    }
```

Comment:
You can simplify this to one line since all you're doing is updating a boolean value in state
```suggestion
    setLengthValidation(signInPassword !== '' && signInPassword.length < minLength)
```

Reply:
simplified the handleValidation function